ambry-test-utils/src/main/java/com/github/ambry/router/InMemoryRouter.java

```diff
@@ -335,15 +335,13 @@ public void setLifeVersion(short lifeVersion) {
       checkBlobId(blobId);
       if (!blobs.containsKey(blobId)) {
         exception = new RouterException("Blob not found", RouterErrorCode.BlobDoesNotExist);
-      } else if (undeletedBlobs.contains(blobId)) {
-        exception = new RouterException("Blob already undeleted", RouterErrorCode.BlobUndeleted);
       } else if (!deletedBlobs.contains(blobId)) {
         exception = new RouterException("Blob not deleted", RouterErrorCode.BlobNotDeleted);
       }
       undeletedBlobs.add(blobId);
       deletedBlobs.remove(blobId);
       if (notificationSystem != null) {
-        notificationSystem.onBlobDeleted(blobId, serviceId, null, null);
+        notificationSystem.onBlobUndeleted(blobId, serviceId, null, null);
```

Comment:
why is this changed to `onBlobUndeleted `?

Reply:
This was a mistake, this is actually undelete function, but I copied the function from delete and forgot to change onBlobDeleted to onBlobUndeleted. Now I am changing it.